### PR TITLE
test: Convert Bech32 test vectors into known-answer test vectors

### PR DESCRIPTION
**Pull request by str4d**
This enables other projects to confirm independently that their encoding
or decoding functions are consistent, instead of merely that they are
round-trip correct.

### CODE REVIEW

File: src/test/bech32_tests.cpp [hunk 1]
```diff
@@ -24,19 +25,43 @@ bool CaseInsensitiveEqual(const std::string &s1, const std::string &s2)
...
+        },
+        {
+            "split1checkupstagehandshakeupstreamerranterredcaperred2y9e3w",
+            "split",
+            "c5f38b70305f519bf66d85fb6cf03058f3dde463ecd7918f2dc743918f2d",
+        },
+        {"?1ezyfcl", "?", ""},
     };
-    for (const std::string& str : CASES) {
+    for (const auto& [str, hrp, data] : CASES) {
```
Comment (str4d) [inline src/test/bech32_tests.cpp:54]: Ahh, C++17 🥰

**Review by daira — APPROVED** [review]
utACK

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
@zkbot retry

**Pull request merged**